src/components/BusinessTable/BusinessTable.jsx

```diff
@@ -122,152 +157,301 @@ const BusinessTable = businessData => {
       } catch (error) {
         console.error('Error sending reminders:', error);
       }
     }
+    const message = `To ${selectedBusinessIds.size} ${
+      selectedBusinessIds.size > 1 ? `businesses` : ` business`
+    }.`;
+    toast({
+      title: 'Reminder Sent',
+      description: message,
+      status: 'success',
+      duration: 3000,
+      isClosable: true,
+    });
   };
 
-
-
-
   const handleDownloadCSV = () => {
     const ids = Array.from(selectedBusinessIds);
-    var headers = [];
+    let headers = [];
     for (var i = 0; i < TABLE_HEADERS.length; i++) {
       headers.push(TABLE_HEADERS[i].toLowerCase().replace(' ', '_'));
     }
-    DownloadCSV(headers, ids, 'business');
+    try {
+      DownloadCSV(ids);
+      const message = `For ${ids.length} ${ids.length > 1 ? `businesses` : ` business`}.`;
+      toast({
+        title: 'Downloaded CSV',
+        description: message,
+        status: 'success',
+        duration: 3000,
+        isClosable: true,
+      });
+    } catch (error) {
+      toast({
+        title: 'Error Downloading CSV',
+        description: error.message,
+        status: 'success',
+        duration: 3000,
+        isClosable: true,
+      });
+    }
+  };
+
+  const getResidentialStatusBadge = status => {
+    if (status === 'Pending') {
+      return (
+        <Badge colorScheme="orange" px="2">
+          <Flex gap={1}>
+            <Text>Pending</Text>
+          </Flex>
+        </Badge>
+      );
+    }
+    if (status === 'Residential') {
+      return (
+        <Badge colorScheme="green" px="2">
+          <Flex gap={1}>
+            <Text>Residential</Text>
+          </Flex>
+        </Badge>
+      );
+    }
+    return (
+      <Badge colorScheme="red" px="2">
+        <Flex gap={1}>
+          <Text>Non-Residential</Text>
+        </Flex>
+      </Badge>
+    );
   };
 
   useEffect(() => {
     const getData = async () => {
       try {
-        setData(businessData['businessData']);
+        const searchTerm = search.replace(' ', '+');
+        const businessResponse = await backend.get(
+          `/business/filter/${currentTab}?pageLimit=10&pageNum=${currentPageNum}&searchTerm=${searchTerm}`,
+        );
+        const businessCountResponse = await backend.get(
+          `/business/totalBusinesses?tab=${currentTab}&searchTerm=${searchTerm}`,
+        );
+        if (currentTab === 'All' && search === '' && businessCountResponse.data[0]['count'] === 0) {
+          onOpen();
+        }
+        setPageLimit(Math.ceil(businessCountResponse.data[0]['count'] / 10));
+        setCurrentBusinessNum(businessCountResponse.data[0]['count']);
+        setData(businessResponse.data);
+        if (currentTab === 'Pending') {
+          setHeaders(PENDING_HEADERS);
+        }
       } catch (error) {
         console.error('Error fetching data:', error);
       }
     };
     getData();
-  }, [businessData]);
+  }, [search, currentTab, currentPageNum, backend]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'PENDING_HEADERS' and 'onOpen'. Either include them or remove the dependency array.